data_model/src/asset.rs

```diff
@@ -413,7 +432,7 @@ impl DefinitionId {
     #[inline]
     pub fn new(name: &str, domain_name: &str) -> Result<Self, ParseError> {
```

Comment:
I think it should looks like `Id::new()`, so it should accept `Name` and `DomainId`